lib/galaxy/objectstore/s3.py

```diff
@@ -134,7 +141,12 @@ def _config_to_dict(self):
                 "size": self.cache_size,
                 "path": self.staging_path,
             },
-            "enable_cache_monitor": False,
+            "cache_monitor": {
+                "enabled": self.cache_monitor_enabled,
+                "cache_limit": self.cache_monitor_cache_limit,
+                "interval": self.cache_monitor_interval,
+                "startup_delay": self.cache_monitor_startup_delay,
+            },
```

Comment:
```suggestion
    "cache_monitor": {"enabled": False}
```

We do not want to serialize the cache manager, it's not a coincidence we set this to False

Reply:
Do you mean for extended metadata @mvdbeek ? I'm not sure that I agree this is the right way to handle that - I feel like we should write the correct configuration but boot up the object store in such a way that we indicate not to cache results? Otherwise all these methods should be renamed "serialize_for_jobs" or something more specific right? Or.... am I missing other use cases / reasons?

Reply:
Here is how I would prefer we disable the cache monitoring for extended metadata - rather than rewriting the configuration of the object store just create it with an application configuration that disables running these kinds of threads. https://github.com/galaxyproject/galaxy/commit/ec041443a9b51bd768e20a96bec09c145ff70217

This seems like a better separation of concerns - the object store shouldn't need to know how to serialize itself in a particular way for set_metadata - especially because none of the code in there makes it clear it is about metadata execution. It is better just to serialize itself with its actual configuration and allow broadly speaking any consume to disable these threads if they need to.